htdocs/compta/facture/card.php

```diff
@@ -3005,6 +3008,11 @@
 		$mode_reglement_id = GETPOST("mode_reglement_id", 'int');
 	}
 
+	// when bank account is empty (means not override by payment mode form a other object, like third-party), try to use default value
+	if (empty($fk_account)) {
```

Comment:
i think you should replace this test and line with
$fk_account = GETPOSTISSET("fk_account") ? GETPOST("fk_account", 'int') : $fk_account;
Because priority is always the value coming from the POST form and then other default values.